library/agent/ServiceConfig.ts

```diff
@@ -2,23 +2,34 @@ import { IPMatcher } from "../helpers/ip-matcher/IPMatcher";
 import { LimitedContext, matchEndpoints } from "../helpers/matchEndpoints";
 import { isPrivateIP } from "../vulnerabilities/ssrf/isPrivateIP";
 import { Endpoint } from "./Config";
-import { IPList } from "./api/fetchBlockedLists";
+import { AgentBlockList, IPList } from "./api/fetchBlockedLists";
 
 export class ServiceConfig {
   private blockedUserIds: Map<string, string> = new Map();
   // IP addresses that are allowed to bypass rate limiting, attack blocking, etc.
   private bypassedIPAddresses: IPMatcher | undefined;
   private nonGraphQLEndpoints: Endpoint[] = [];
   private graphqlFields: Endpoint[] = [];
-  private blockedIPAddresses: { blocklist: IPMatcher; description: string }[] =
-    [];
+  private blockedIPAddresses: {
+    key: string;
+    blocklist: IPMatcher;
+    description: string;
+  }[] = [];
   private blockedUserAgentRegex: RegExp | undefined;
+  private monitoredUserAgents: {
+    key: string;
+    pattern: RegExp;
+  }[] = [];
```

Comment:
No strong opinion about that, but do we even need two different lists? We could add a property to the existing list entries. Or are other agents validating additional keys?